test/indexer/__init__.py

```diff
@@ -136,3 +136,31 @@ def mocked_get_bundle(bundle_uuid, bundle_version):
             with patch.object(indexer, '_get_bundle', new=mocked_get_bundle):
                 contributions = indexer.transform(notification, delete=False)
                 return indexer.contribute(contributions)
+
+    def _verify_sorted_lists(self, value, path):
```

Comment:
Please add type hints

Reply:
Path should be a tuple and have a default of `()`

Reply:
```suggestion
    def _assert_sorted_leaf_lists(self, value, path):
```

Reply:
Scratch the default argument for path. I would like the bulk of the body of this method to move into a nested function so the path argument to the method disappears completely and the return value becomes None. The assertion of `> 0` becomes part of the method and can be eliminated from the two call sites. That simplifies the signature and you have to explain less in the docstring.